WordPress/Classes/ViewRelated/Menus/Views/MenusSelectionView.h

```diff
@@ -0,0 +1,74 @@
+#import <UIKit/UIKit.h>
+#import "MenusSelectionItem.h"
+
+typedef NS_ENUM(NSUInteger, MenusSelectionViewType) {
+    MenusSelectionViewTypeMenus = 1,
+    MenusSelectionViewTypeLocations
+};
+
+@protocol MenusSelectionViewDelegate;
+
+@interface MenusSelectionView : UIView
+
+@property (nonatomic, weak) id <MenusSelectionViewDelegate> delegate;
+
+/**
+ The type of selection the selectionView should be configured for.
+ */
+@property (nonatomic, assign) MenusSelectionViewType selectionType;
+
+/**
+ Toggle the visual expansion of the selection items in the UI.
+ */
+@property (nonatomic, assign) BOOL selectionItemsExpanded;
+
+/**
+ The currently selected item in the UI.
+ */
+@property (nonatomic, strong) MenusSelectionItem *selectedItem;
+
+/**
+ Add a selection item to the list of available items for display.
+ */
+- (void)addSelectionViewItem:(MenusSelectionItem *)selectionItem;
+
+/**
+ Remove a selection item from the list of available items for display.
+ */
+- (void)removeSelectionItem:(MenusSelectionItem *)selectionItem;
+
+/**
+ Remove all selection items from the list of available items for display.
+ */
+- (void)removeAllSelectionItems;
+
+/**
+ Create a new selection item with an associated object the item represents.
+ */
+- (MenusSelectionItem *)itemWithItemObjectEqualTo:(id)itemObject;
```

Comment:
For these, I wanted to keep the `MenusSelectionItem` representing a generic object with `@property (nonatomic, strong) id itemObject;` instead of a specific class/object. This would be useful as we've gone a bit back and forth over showing different selection types here that may or may not be managed objects.